src/argilla_sdk/records/_dataset_records.py

```diff
@@ -212,7 +212,10 @@ def add(
         Add generic records to a dataset as dictionaries:
 
         """
-        record_models = self.__ingest_records(records=records, mapping=mapping, user_id=user_id or self.__client.me.id)
+        if isinstance(records, HFDataset):
+            records = self()._record_dicts_from_datasets(records)
```

Comment:
IF we move this somehow to the `_ingest_records` we will have the feat. of adding and updating records from datasets for free.